src/Diagnostics.HealthChecks/DependencyInjection/OmexHealthChecksBuilderAddCheckExtensions.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable IDE0008 // Use explicit type
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Omex.Extensions.Diagnostics.HealthChecks;
+using DynamicallyAccessedMembersAttribute = Microsoft.Omex.Extensions.Diagnostics.HealthChecks.DynamicallyAccessedMembersAttribute;
+using DynamicallyAccessedMemberTypes = Microsoft.Omex.Extensions.Diagnostics.HealthChecks.DynamicallyAccessedMemberTypes;
+using UnconditionalSuppressMessageAttribute = Microsoft.Omex.Extensions.Diagnostics.HealthChecks.UnconditionalSuppressMessageAttribute;
+
+namespace Microsoft.Omex.Extensions.DependencyInjection;
+
+/// <summary>
+/// Provides basic extension methods for registering <see cref="IHealthCheck"/> instances in an <see cref="IOmexHealthChecksBuilder"/>.
+/// </summary>
+public static class OmexHealthChecksBuilderAddCheckExtensions
+{
+	/// <summary>
+	/// Adds a new health check with the specified name and implementation.
+	/// </summary>
+	/// <param name="builder">The <see cref="IOmexHealthChecksBuilder"/>.</param>
+	/// <param name="name">The name of the health check.</param>
+	/// <param name="instance">An <see cref="IHealthCheck"/> instance.</param>
+	/// <param name="failureStatus">
+	/// The <see cref="HealthStatus"/> that should be reported when the health check reports a failure. If the provided value
+	/// is <c>null</c>, then <see cref="HealthStatus.Unhealthy"/> will be reported.
+	/// </param>
+	/// <param name="tags">A list of tags that can be used to filter health checks.</param>
+	/// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+	/// <param name="parameters">An optional <see cref="HealthCheckRegistrationParameters"/> representing the individual health check options.</param>
+	/// <returns>The <see cref="IOmexHealthChecksBuilder"/>.</returns>
+	[SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required to maintain compatibility")]
+	public static IOmexHealthChecksBuilder AddCheck(
+		this IOmexHealthChecksBuilder builder,
+		string name,
+		IHealthCheck instance,
+		HealthStatus? failureStatus = default,
+		IEnumerable<string>? tags = default,
+		TimeSpan? timeout = default,
+		HealthCheckRegistrationParameters? parameters = default)
+	{
+		if (builder == null)
```

Comment:
any reason not to write `_ = ?? throw new ArgumentNullException(nameof(builder))`?